server/src/server.js

```diff
@@ -33,7 +33,40 @@ Router.run(routes.routes, Router.HistoryLocation, function (Handler, state) {
     gotData(Handler, _d);
     return;
   }
-  console.error("Error: partial page refresh not currently implemented");
+
+  var appnames = [],
+    app = null;
+  for (var i in state.routes) {
+    if (!!state.routes[i].name) {
+      appnames.push(state.routes[i].name);
+      if (app === null) {
+        app = state.routes[i];
+      }
+    }
+  }
+  if (!appnames.length) {
+    console.error("Error: No app was routed");
+    return;
+  }
+
+  console.log("Route to app", appnames[0]);
+
+  if (appnames[0] === "shot") {
```

Comment:
I don't think it's safe to assume the cachedData from the full page load is applicable to the partial page refresh, although right now it's safe because we don't link to any other pages.